parade/parade_test.go

```diff
@@ -133,6 +133,29 @@ func intAddr(i int) *int {
 	return &i
 }
 
+func scanIDs(t *testing.T, prefix string) []parade.TaskID {
+	t.Helper()
+	rows, err := db.Query(`SELECT id FROM tasks WHERE id LIKE format('%s%%', $1::text)`, prefix)
+	if err != nil {
+		t.Errorf("[I] select remaining IDs for prefix %s: %s", prefix, err)
+	}
+
+	defer func() {
+		if err := rows.Close(); err != nil {
+			t.Fatalf("[I] remaining ids iterator close: %s", err)
+		}
+	}()
+	gotIDs := make([]parade.TaskID, 0)
+	for rows.Next() {
+		var id parade.TaskID
+		if err = rows.Scan(&id); err != nil {
+			t.Errorf("[I] scan ID value: %s", err)
```

Comment:
Fatalf - not sure you can continue to scan after that

Reply:
Why not?  It's just a broken row (e.g. parse failure).  `rows.Next` is welcome to fail if the database is :fire:  on fire.

So you're actually right, the code is wrong, and I added a test that rows were successfully scanned.

Reply:
You are right that calling Next after Scan that failed will probably exit. The code reader that sees Errorf will think that this error is something we can live with and continue - in this case adding the id to the slice, which is not what we wanted.